joblib/numpy_pickle.py

```diff
@@ -267,19 +267,14 @@ def save(self, obj):
                     # Pickling doesn't work with memmaped arrays
                     obj = self.np.asarray(obj)
                 return Pickler.save(self, obj)
-            self._npy_counter += 1
-            try:
+
+            if not obj.dtype.hasobject:
+                self._npy_counter += 1
                 filename = '%s_%02i.npy' % (self._filename,
                                             self._npy_counter)
                 # This converts the array in a container
                 obj, filename = self._write_array(obj, filename)
                 self._filenames.append(filename)
-            except:
```

Comment:
@GaelVaroquaux do you remember why you wrote this try/except ? What kind of errors is it supposed to tackle ?